### PR TITLE
Minor Kotlin vs Java formatting improvements

### PR DESCRIPTION
**Pull request by cadamini**
Closes https://github.com/utkarsh006/Eazy-Android/issues/47

- add an h1 element to show the article topic
- use h2 elements to be able to differential the list in step 4
- fix list formatting
- consistently use the NPE acronym after introducing NullPointerException

Note: I have checked the other articles in https://github.com/utkarsh006/Eazy-Android/tree/main/Articles - and they inconsistently use h1 and h2 elements. I decided on h1 and h2 headings. You finally need to find a common way to format headings in all articles.

### CODE REVIEW

File: Articles/1_kotlin_over_java.md [hunk 1]
```diff
@@ -1,47 +1,41 @@
 In many ways, Kotlin designed by JetBrains has become the preferred language for the Android development industry. While it cannot outdo Java in terms of its popularity, many developers emphasise Kotlin will soon become the dominant platform for Android development.
 
+## Kotlin over Java
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:3]:
```suggestion
# Why Kotlin over Java?

In many ways, Kotlin designed by JetBrains has become the preferred language for the Android development industry. While it cannot outdo Java in terms of its popularity, many developers emphasise Kotlin will soon become the dominant platform for Android development.

```

I have checked other articles in https://github.com/utkarsh006/Eazy-Android/tree/main/Articles - and they inconsistently use h1 and h2 elements. This h1 would be another option.

File: Articles/1_kotlin_over_java.md [hunk 2]
```diff
@@ -1,47 +1,41 @@
 In many ways, Kotlin designed by JetBrains has become the preferred language for the Android development industry. While it cannot outdo Java in terms of its popularity, many developers emphasise Kotlin will soon become the dominant platform for Android development.
 
+## Kotlin over Java
+
 When it comes to its usability, Kotlin has a bunch of tools and frameworks that work with Java. Also, **its advanced compiler can check the errors both during compile time and run time**. It also brings down the number of lines of code significantly. Kotlin is officially supported by Google for mobile development on Android since the release of Android Studio 3.0 in October 2017.
 
 Here are 5 reasons why developers like Kotlin over Java and think that it will overtake the later:
 
-**1. Shorter program for the same task**
+### 1. Shorter program for the same task
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:9]:
```suggestion
## 1. Shorter program for the same task

```

File: Articles/1_kotlin_over_java.md [hunk 3]
```diff
@@ -1,47 +1,41 @@
...
 
 Here are 5 reasons why developers like Kotlin over Java and think that it will overtake the later:
 
-**1. Shorter program for the same task**
+### 1. Shorter program for the same task
 
 Kotlin is a statically-typed language which is very easy to read and write. It has a much simpler and shorter code than Java's code for the same problem. As this makes the language more human-readable, it becomes easy to debug. **Kotlin's code is much smaller and streamlines the programming process, in comparison to Java**. This is partly because of Kotlin's slick IDE.
 
-**2. Easy Code**
+### 2. Easy Code
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:13]:
```suggestion
## 2. Easy Code

```

File: Articles/1_kotlin_over_java.md [hunk 4]
```diff
@@ -1,47 +1,41 @@
...
 
 Kotlin programs do not need semicolons in their program. This makes the programs easy to read and understand. They also have smart casts and string templates. **Java is not a succinct language. Such a language code increases the chances of bugs**. The code being in a concise language means fewer chances of both runtime and compile time errors. Kotlin gives a simple way to use mutable and immutable declarations for different data structures.
 
-**3. Java Compatibility**
+### 3. Java Compatibility
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:17]:
```suggestion
## 3. Java Compatibility

```

File: Articles/1_kotlin_over_java.md [hunk 5]
```diff
@@ -1,47 +1,41 @@
...
-**3. Some data inconsistency with regard to initialization, such as when:**
-
-- An uninitialized this available in a constructor is passed and used somewhere ("leaking this");
-
-- A superclass constructor calls an open member whose implementation in the derived class uses uninitialized state;
-
-**4. Java interoperation:**
-
-- Attempts to access a member on a null reference of a platform type;
+### 4. Eliminating Null References
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:23]:
```suggestion
## 4. Eliminating Null References

```

File: Articles/1_kotlin_over_java.md [hunk 6]
```diff
@@ -1,47 +1,41 @@
...
+3. **Some data inconsistency with regard to initialization, such as when:**
+    - An uninitialized this available in a constructor is passed and used somewhere ("leaking this");
+    - A superclass constructor calls an open member whose implementation in the derived class uses uninitialized state;
+4. **Java interoperation:**
+    - Attempts to access a member on a null reference of a platform type;
+    - Generic types used for Java interoperation with incorrect nullability, e.g. a piece of Java code might add null into a Kotlin MutableList\<String\>, meaning that MutableList\<String?\> should be used for working with it;
+    - Other issues caused by external Java code.
 
-**5. Solution To Some Of Java's Flaws**
+### 5. Solution To Some Of Java's Flaws
```
Comment (cadamini) [inline Articles/1_kotlin_over_java.md:37]:
```suggestion
## 5. Solution To Some Of Java's Flaws

```

**Comment by utkarsh006** [comment]
LGTM ⭐

**Pull request merged**